src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -311,6 +358,14 @@ private int recordExitStatus(FilePath workspace, int code) throws IOException, I
         private static final long serialVersionUID = 1L;
     }
 
+    private static final class getArchitecture extends MasterToSlaveCallable<String,RuntimeException> {
+        @Override public String call() throws RuntimeException {
+            // Note: This will only determine the architecture of the JVM.
+            return System.getProperty("sun.arch.data.model");
```

Comment:
Is this specific to Oracle/OpenJDK? I would move the lookup here into the same callable as `getOsType`, change it to `getOsAndArch` or something and return both values simultaneously so we don't have to make two separate calls to the remote agent in quick succession.

If you only use the JVM arch, why not use the `os.arch` property instead? Just from the name seems more likely to be supported across JVMs. Did you do any testing to see if 32-bit Go binaries work on 64-bit platforms when a 32-bit JVM is in use?

Reply:
So this property is apparently also available on IBM jvm as well, but I did not specifically check.  The thing about using os.arch is that it is meant only to be for human eyes.  There are actually a good amount of values that could be populated in there.
see:
https://stackoverflow.com/questions/2062020/how-can-i-tell-if-im-running-in-64-bit-jvm-or-32-bit-jvm-from-within-a-program#comment25544753_2062045